src/System.Runtime.Extensions/tests/VoidMainWithExitCodeApp/VoidMainWithExitCodeApp.cs

```diff
@@ -4,16 +4,39 @@
 
 using System;
 using System.Reflection;
+using System.Threading;
 
 namespace VoidMainWithExitCodeApp
 {
     internal static class Program
     {
         static void Main(string[] args)
         {
-            int exitCode = args.Length > 0 ? int.Parse(args[0]) : 0;
-            typeof(Environment).GetTypeInfo().GetDeclaredProperty("ExitCode").SetValue(null, exitCode);
-            // Environment.ExitCode = exitCode; // TODO: Remove reflection when package updated with latest Environment exposing ExitCode
+            int exitCode = int.Parse(args[0]);
+            int mode = int.Parse(args[1]);
+
+            PropertyInfo set_ExitCode = typeof(Environment).GetTypeInfo().GetDeclaredProperty("ExitCode");
+            MethodInfo Exit = typeof(Environment).GetTypeInfo().GetDeclaredMethod("Exit");
+
+            switch (mode)
+            {
+                case 1: // set ExitCode and exit
+                    set_ExitCode.SetValue(null, exitCode); // TODO: Environment.ExitCode = exitCode;
+                    break;
+
+                case 2: // set ExitCode, exit, and then set ExitCode from another foreground thread
+                    new Thread(() => // foreground thread
+                    {
+                        Thread.Sleep(1000); // time for Main to exit
```

Comment:
Is it possible to use an event? Or is the idea that the app will not wait to terminate (since generally background threads don't prevent terminate -- in which case, I guess this is randomly doing something, or not, depending on timing)